src/btree/bt_debug.c

```diff
@@ -1214,55 +1213,69 @@ __debug_page_metadata(WT_DBG *ds, WT_REF *ref)
         return (__wt_illegal_value(session, page->type));
     }
 
-    WT_RET(ds->f(ds, ": %s\n", __wt_page_type_string(page->type)));
-    WT_RET(__debug_ref(ds, ref));
-
-    WT_RET(ds->f(ds,
-      "\t"
-      "disk %p",
-      (void *)page->dsk));
+    WT_RET(ds->f(ds, "disk_img: %p", (void *)page->dsk));
     if (page->dsk != NULL)
-        WT_RET(ds->f(ds, ", dsk_mem_size %" PRIu32 ", write_gen: %" PRIu64, page->dsk->mem_size,
+        WT_RET(ds->f(ds, " | dsk_mem_size: %" PRIu32 " | write_gen: %" PRIu64, page->dsk->mem_size,
           page->dsk->write_gen));
-    WT_RET(ds->f(ds, ", entries %" PRIu32, entries));
-    WT_RET(ds->f(ds, ", %s", __wt_page_is_modified(page) ? "dirty" : "clean"));
-
-    if (F_ISSET_ATOMIC_16(page, WT_PAGE_BUILD_KEYS))
-        WT_RET(ds->f(ds, ", keys-built"));
-    if (F_ISSET_ATOMIC_16(page, WT_PAGE_DISK_ALLOC))
-        WT_RET(ds->f(ds, ", disk-alloc"));
-    if (F_ISSET_ATOMIC_16(page, WT_PAGE_DISK_MAPPED))
-        WT_RET(ds->f(ds, ", disk-mapped"));
-    if (F_ISSET_ATOMIC_16(page, WT_PAGE_EVICT_LRU))
-        WT_RET(ds->f(ds, ", evict-lru"));
-    if (F_ISSET_ATOMIC_16(page, WT_PAGE_INTL_OVERFLOW_KEYS))
-        WT_RET(ds->f(ds, ", overflow-keys"));
-    if (F_ISSET_ATOMIC_16(page, WT_PAGE_SPLIT_INSERT))
-        WT_RET(ds->f(ds, ", split-insert"));
-    if (F_ISSET_ATOMIC_16(page, WT_PAGE_UPDATE_IGNORE))
-        WT_RET(ds->f(ds, ", update-ignore"));
+    WT_RET(ds->f(ds, " | entries: %" PRIu32, entries));
+    WT_RET(ds->f(ds, " | state: %s", __wt_page_is_modified(page) ? "dirty" : "clean"));
+
+    if (page->flags_atomic != 0) {
```

Comment:
I appreciate this maintains the structure of the pre-existing code. I think the repetitive code in the flag dumping can be simplified.

Stating the problem as I read the code: output text name for each set flag with a comma between each flag.

Here's a very rough outline of the idea:
```
int mask_count = 4;
const u64 masks[] = { MSK_1, MSK_2, MSK_3, MSK_4 };
const char *names[] = { "#1", "#2", "#3", "#4" };

char *out[mask_count];
int out_count = 0;
for (int i = 0; i < mask_count; i++) {
  if (flags & masks[i]) {
    out[i] = names[i];
    out_count++;
  } else {
    out[i] = NULL;
  }
}
for (int j = 0; j < mask_count && out_count > 0; j++) {
  if (out[i]) { print(out[i]); out_count--; }
  if (out_count > 0) print(", ");
}
```

Suspect bugs in this example, happy to discuss IRL.